### ISSUE TITLE
Exodii Mission 3: Dealing with the Free Merchants

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#59911]
### Is your feature request related to a problem? Please describe.

See https://github.com/CleverRaven/Cataclysm-DDA/issues/59900

The Exodii missions haven't yet been implemented, and I am worried that my life commitments will keep them out of my reach for a long time. I am posting my plan for their missions in the hopes that someone will help me implement parts of them. If that happens, I would be more than happy to contribute with dialogue and editing and some of the components as I have time.

Of course I may get to this myself, in which case having it written up will keep me from forgetting what I wanted.

### Solution you would like.

This is as far as I've gotten on planning Exodii missions so far, but it does a very good job of setting the stage for the faction and modelling some of the more complex missions and dialogue we can do now.

This is the initial version of this mission, but it is still a relatively complex mission to implement even without taking into account all the different ways you can achieve it. Most of the refinements will actually be about world changes following the results of this mission.

#### If you have pissed off the exodii...
If you are on the "cautious" or "trusted" mission pathways for the exodii, you can do this mission, it's not significantly different either way but Rubik will have different introductory dialogue if they are cautious about you. If you are considered "dangerous" then, for now, you won't get any further mission options (later we will outline a separate similar mission if the exodii don't trust you).

#### Setup:
- The Exodii have become aware of the Free Merchants, and initial information suggests they would be a good trade partner. They would like you to find out more about the merchants, report back to Rubik, and then be their representative in a trade discussion with Smokes.
- They send you with a radio for the discussion so that Rubik can listen in, unless you have a CBM that can do this. This is above board, you are supposed to tell smokes about it but explain the language barrier.

#### Phase 1: Get some information.
- If you have already encountered the Free Merchants, you can immediately skip this stage by sharing what you know with Rubik.
- If you have not met them before, Rubik will give you directions and ask you to report back to them before proceeding, so that they can offer the deal, but make it clear they don't mind if you head in and chat.
- There is no consequence for talking to the Free Merchants before reporting back to Rubik this time. If you tell Smokes you are interested in brokering a deal between them, he'll just be intrigued and ask to speak to them.
- Side dialogue - if you previously approached the Hub against Rubik's advice in mission 1, Rubik will tease you for your inability to follow directions, but acknowledge that it wasn't a big deal this time.

#### Phase 2: Set up a meeting.
- Rubik will give you a radio if you don't have one, unless you have some sort of radio communication CBM (I don't htink we have one yet do we?)
- Rubik primes you on the terms they are looking for and what they are willing to trade.
    - The exodii are looking for plastics, refined steel, and other bulk goods, negotiable individually. Specifically they are not looking for wood, basic construction materials, fuel, or most food products. They believe that their needs are often for things that are less useful to other survivors.
    - Rubik gives you permission to explain who and what they are in terms that are understandable in your world.
    - In return for first trade rights to some of the goods they are looking for, they'll offer not only trade on site, but caravan protection within a radius of their base, and safe shelter for Free Merchants looking for stopover housing in the area.
    - You can ask Rubik if there's anything further they're willing to share if asked; they say that if it reaches that point, they can't authorize you to make further deals and will have to take over.
- You can start a trade meeting through a discussion with smokes
- Ideally, we should have you and Smokes move to a room with the Free Merchant broker to have the meeting, but I don't think that is currently easy enough to do with JSON, so for now you can just talk to him in the lobby.

#### Phase 3: The meeting.
- You activate the radio with Rubik, who introduces themselves to Smokes and vice versa and then gives you permission to initiate the discussion.
- At each phase of the discussion you have a few, fairly obviously worded options:
    - Favour Exodii, Social: Explain the situation and spin it in favour of the exodii
    - Favour Merchants, Social: Explain the situation and spin it in favour of the merchants
    - Play dumb: Make a neutral statement that doesn't directly further the discussion
    - Maverick: Aggravate the situation intentionally.
- We track three variables here: Exodii favoured, Merchants favoured, Incompetence, and Hostility.
    - Passing the 'spin' social check moves the meter in favour of faction you aimed at (+2). Failing it moves the meter in favour of the faction you didn't aim at (+1), increases your Incompetence score (+1), and may increase Hostility at random.
    - The neutral option doesn't move the meter at all in either direction. In some cases this may increase or decrease Incompetence, depending on how you use it.
    - Passing the 'aggravate' check doesn't move the meter but raises hostility. Failing it increases Incompetence +4.
- Occasionally we might offer a fourth dialogue option gated to a specific stat or piece of knowledge/skill you have, giving either an easier persuasion check or a larger bonus to whichever faction you're negotiating in favour of.

The discussion should pass through about 10 stages depending on what dialogue you choose. At the end, the faction with the higher Favour score gets the better deal. The dialogue will change through the stages depending on who has the better deal.
- If your Incompetence score passes 5, Rubik takes over the negotiation. You will get one more opportunity to attempt to screw things up if you for some reason want to burn bridges, and then the default result occurs.
- If hostility increases beyond 5, the negotiation fails and the Free Merchants refuse to deal with the Exodii.

**Incomplete issue:** I will add more here when done, but must take a break.

**Comment by descan** [comment]
Been reading these, they seem neat - not that I can help much, though.

I did want to ask - what's the planned actual impact, both in-world and for the player, on a faction getting more favoured in the deal here? Exodii access to materials and expansion? Merchants getting access to CBMs? Cheaper prices, more trust (less trust for the opposite?), access to different stock at different levels for the two factions?

**Comment by I-am-Erk** [comment]
All in good time. Short version: if the Exodii are favoured, the free merchant traders that spawn near them will have better stock. If the free merchants are favoured, they'll get an autodoc sooner. Probably some other fallout as well

**Comment by PatrikLundell** [comment]
The mission ought to make it clear you are going to do the negotiations, not just bring a radio along, so the player understands the Social skill is going to be used.

I think there's room for two "neutral" stances, one that tries to stay out of the way so as to not bungle it up (for the socially inept), and one that tries to get both parties more interested in reaching an agreement (for those socially gifted), ideally with more trade in both directions that would decrease tension if successful, and increase it if unsuccessful. If provided, it still shouldn't lead to the results of strongly favoring both parties, but rather weakly favoring both (through greater volumes traded, with both parties perceiving they get good deals).

**Comment by I-am-Erk** [comment]
Good point, mostly there just needs to be the secondary mission option that you can tell Rubik you're a poor negotiator, and just introduce them to the Merchants. That will also be basically the same as the Dangerous path mission so it's easy to add.

**Comment by descan** [comment]
I'm not sure if I would call that "highly nuanced outcomes," what they're suggesting. It sounds like they'd want a general range of each faction having three (four?) states - heavily favoured, slightly favoured, no bias, (and maybe disfavoured as the fourth?) with Heavily Favoured being exclusive to a deal biased in their direction, rather than something you can achieve by trying to get everyone to get along. And that you could get Slightly Favoured status for both by trying to Give-Peace-A-Chance it.

There can be some nuance there, sure, but on first pass they don't have to intermesh all that much. (I.e. you *could* have a situation of Heavily Favoured Merchants and Slightly Favoured Exodii that changes the flavour or substance of the Heavily Favoured merchants outcome compared to Heavily Favoured merchants and No-Bias Exodii, but first pass wouldn't need to have that variable checked, I don't think?)

**Comment by I-am-Erk** [comment]
That's a switch from three outcome states (exodii deal, merchant deal, no deal) to around sixteen.

**Comment by descan** [comment]
I suppose. I don't know how it would be coded. I was thinking it'd be more like 8 states, most of which could be concurrent on different overmap tiles, although I can see how it would require more nuanced coding if they were taking place on the *same* overmap tile. (needing a version of the tile with both states at the same time as well as one for each lonesome state, turning 2 states into 3 states.) My bad!

**Comment by I-am-Erk** [comment]
It's more that it requires a ton of further error trapping in each stage down the road. Did you secure an agreement to get an autodoc for the free merchants, or not? Are the free merchants giving the exodii exclusivity on certain trade items, or not? If these are relevant to gameplay then many of them will probably affect future outcomes, and if they're not relevant then we don't really need to monitor them too closely and can just handwave differences with a bit of dialogue. I'm open to having flavour text change depending on specific details of the interaction, but the more important the differences are the more they could shape the outcomes of further missions, and I want to avoid that where i can.

**Comment by I-am-Erk** [comment]
Ooh. I forgot how fun and detailed I had made this. It doesn't need as much thought as I remembered it needing.
